src/conn/conn_sweep.c

```diff
@@ -6,13 +6,48 @@
  * See the file LICENSE for redistribution information.
  */
 
+#include "wiredtiger.h"
 #include "wt_internal.h"
 
 #define WT_DHANDLE_CAN_DISCARD(dhandle)                           \
     (!F_ISSET(dhandle, WT_DHANDLE_EXCLUSIVE | WT_DHANDLE_OPEN) && \
       __wt_atomic_loadi32(&(dhandle)->session_inuse) == 0 &&      \
       __wt_atomic_load32(&(dhandle)->references) == 0)
 
+/*
+ * __sweep_check_file_handle_exists --
+ *     Check if the file dhandle exists for the table dhandle.
+ */
+static int
+__sweep_check_file_handle_exists(WT_SESSION_IMPL *session, WT_DATA_HANDLE *dhandle)
+{
+    WT_DECL_RET;
+    WT_TABLE *table;
+
+    ret = WT_NOTFOUND;
+    /*
+     * The sweep server's algorithm is altered to prevent unnecessary table dhandle closures. This
+     * is done by checking for associated file dhandles before marking table dhandles for sweeping.
+     * It resolves schema lock contention caused by repetitive table dhandle operations during
+     * MongoDB cursor activity on simple tables, and ensures table dhandles are retained for active
+     * file dhandles, which is required for file dhandle access.
+     */
+    table = (WT_TABLE *)dhandle;
+    if (table->is_simple && table->cg_complete) {
+        ret = __wt_conn_dhandle_find(session, table->cgroups[0]->source, NULL);
```

Comment:
I would think the timeofdeath would be the same for both `table:` and `file:` handles?